packages/flutter_tools/lib/src/commands/migrate.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
+import '../cache.dart';
+import '../migrate/migrate_utils.dart';
+import '../runner/flutter_command.dart';
+import 'migrate_abandon.dart';
+import 'migrate_apply.dart';
+import 'migrate_resolve_conflicts.dart';
+import 'migrate_start.dart';
+import 'migrate_status.dart';
+
+const String kDefaultMigrateWorkingDirectoryName = 'migrate_working_dir';
+
+/// Base command for the migration tool.
+class MigrateCommand extends FlutterCommand {
+  MigrateCommand({
+    bool verbose = false,
+    required this.logger,
+    required FileSystem fileSystem,
+    required Terminal terminal,
+  }) : _verbose = verbose {
+    addSubcommand(MigrateAbandonCommand(verbose: _verbose, logger: logger, fileSystem: fileSystem));
+    addSubcommand(MigrateApplyCommand(verbose: _verbose, logger: logger, fileSystem: fileSystem));
+    addSubcommand(MigrateResolveConflictsCommand(verbose: _verbose, logger: logger, fileSystem: fileSystem, terminal: terminal));
+    addSubcommand(MigrateStartCommand(verbose: _verbose, logger: logger, fileSystem: fileSystem));
+    addSubcommand(MigrateStatusCommand(verbose: _verbose, logger: logger, fileSystem: fileSystem));
+    // TODO(garyq): add command for guided conflict resolution.
+  }
+
+  final bool _verbose;
```

Comment:
I'm pretty sure you don't need to actually track these in your Commands. Any time you want verbose logging, you can call `logger.printTrace()`, and it will only actually show in the logger if the logger in the context is verbose. See https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/executable.dart#L126